dashboard/app/assets/javascripts/header.js

```diff
@@ -415,35 +412,48 @@ dashboard.header.updateTimestamp = function () {
 };
 
 /**
- * Get the user progress for an anonymous user from the client side cookie.
- * @param {string} scriptName The script to get progress for.
- * @return Object that is a representation of the user's individual level progress.
+ * See ApplicationHelper#activity_css_class.
+ * @param result
+ * @return {string}
  */
-function getSummarizedProgressForAnonymousUser (scriptName) {
-  var summarizedProgress = {};
-  var levelProgress = {};
+function activityCssClass(result) {
+  if (!result) return 'not_tried';
+  if (result >= 1000) return 'submitted';
+  if (result >= 30) return 'perfect';
+  if (result >= 20) return 'passed';
+  return 'attempted';
+}
 
-  summarizedProgress.lines = dashboard.clientState.lines();
+/**
+ * Returns the "best" of the two results, as defined in apps/src/constants.js.
+ * Note that there are negative results that count as an attempt, so we can't
+ * just take the maximum.
+ * @param {Number} a
+ * @param {Number} b
+ * @return {string} The result css class.
+ */
+function mergeProgress(a, b) {
+  a = a || 0;
+  b = b || 0;
+  if (a === 0) {
+    return activityCssClass(b);
+  }
+  if (b === 0) {
+    return activityCssClass(a);
+  }
+  return activityCssClass(Math.max(a, b));
+}
 
+function populateProgress(scriptName) {
   var scriptProgress = dashboard.clientState.allLevelsProgress()[scriptName] || {};
-  for (var level in scriptProgress) {
-    levelProgress[level] = {};
-    var individualLevelProgress = scriptProgress[level] || -1;
-
-    if (individualLevelProgress >= 1000) {
-      levelProgress[level].status = 'submitted';
-    } else if (individualLevelProgress >= 30) {
-      levelProgress[level].status = 'perfect';
-    } else if (individualLevelProgress >= 20) {
-      levelProgress[level].status = 'passed';
-    } else if (individualLevelProgress > 0) {
-      levelProgress[level].status = 'attempted';
-    } else {
-      levelProgress[level].status = 'not attempted';
-    }
-  }
 
-  summarizedProgress.levels = levelProgress;
+  // Aggregate progress from server and client
+  $.each(scriptProgress, function (level_id, result) {
+    var level_link = $('[data-level=' + level_id + ']');
```

Comment:
As discussed let's use an id here rather than a data attribute.  (MIght be more efficient, and is more self-documenting.)